CMakeLists.txt

```diff
@@ -341,6 +341,10 @@ if (DALI_CLANG_ONLY)
 endif()
 
 # Add ptx & bin flags for cuda compiler (nvcc)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16.0")
```

Comment:
One open question is: should we allow to provide `CMAKE_CUDA_ARCHITECTURES` directly? - as this is the official way of doing it, and the `CUDA_targeted_archs` was our workaround before `CMAKE_CUDA_ARCHITECTURES` was a thing. 

That depends on our philosophy if we want to be fully-CMake compliant or support this as another option, we can:
* detect if `CMAKE_CUDA_ARCHITECTURES` is set and use it with precedence over `CUDA_targeted_archs`
* keep `CUDA_targeted_archs` for all usages, and automatically propagate to  `CMAKE_CUDA_ARCHITECTURES` as in this PR - maybe we should warn the user when he tries to set it by hand?

What's with the documentation of this feature? The `USE_CMAKE_CUDA_ARCHITECTURES` was rather well hidden.

Reply:
How about now?